scene/resources/shader.cpp

```diff
@@ -177,7 +177,16 @@ bool Shader::is_text_shader() const {
 }
 
 bool Shader::has_uniform(const StringName &p_param) const {
-	return params_cache.has("shader_uniform/" + p_param);
+	if (params_cache_dirty) {
+		get_shader_uniform_list(nullptr);
+	}
+	const StringName value = "shader_uniform/" + p_param;
+	for (const auto &d : params_cache) {
```

Comment:
Why do a linear search? (it's more expensive on a map than on a list also)
params_cache is a HashMap with the parameter name as key.